### PR TITLE
Unified search: add attributes for clients

### LINKED ISSUE #3997: Unified Search API Response Update For Contact
### Is your feature request related to a problem? Please describe.

[We want to](https://github.com/nextcloud/android/issues/12679) open contact directly in mobile application rather than redirect to web app. Therefore, we need to use Unified Search API to get contact and try to match with already synced internal contacts by Dav5x. So that user can open contact in phone.

### Describe the solution you'd like

Dav5x contact ID and internal Android contact ID is not matching so we need to provide uniqueness via depending on DISPLAY_NAME, EMAIL, PHONE_NUMBER, UID(if we have on both side even better) 

[VCARD](https://en.wikipedia.org/wiki/VCard)

**Current Implementation**

```
{
"thumbnailUrl": "",
"title": "Test",
"subline": "[dummy@test.de](https://cloud.nextcloud.com/call/mailto:dummy@test.de)",
"resourceUrl": "http://localhost/nc/index.php/apps/contacts/direct/contact/8ccc7b7f-19ee-4f9c-9a70-ab78cfc9ac0c~contacts",
"icon": "icon-contacts-dark",
"rounded": true,
"attributes": []
}
```

**Suggestion**

```
{
"thumbnailUrl": "",
"title": "Test",
"subline": "[dummy@test.de](https://cloud.nextcloud.com/call/mailto:dummy@test.de)",
"resourceUrl": "http://localhost/nc/index.php/apps/contacts/direct/contact/8ccc7b7f-19ee-4f9c-9a70-ab78cfc9ac0c~contacts",
"icon": "icon-contacts-dark",
"rounded": true,
"attributes": [
"displayName": "Arthur Kirkland"
"email": "example@mail.com"
"phoneNumber": 00000000000
]
}
```

**Better Alternative Need To Check For Android**

```
{
"thumbnailUrl": "",
"title": "Test",
"subline": "[dummy@test.de](https://cloud.nextcloud.com/call/mailto:dummy@test.de)",
"resourceUrl": "http://localhost/nc/index.php/apps/contacts/direct/contact/8ccc7b7f-19ee-4f9c-9a70-ab78cfc9ac0c~contacts",
"icon": "icon-contacts-dark",
"rounded": true,
"attributes": [
"uid": "4fbe8971-0bc3-424c-9c26-36c3e1eff6b1"
]
}
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #6079: Unified Search API Response Update For Calendar Event
[We want to](https://github.com/nextcloud/android/issues/12679) open calendar events directly in mobile application rather than redirect to web app. Therefore, we need to use Unified Search API to get calendar events and try to match with already synced internal calendar events by Dav5x. So that user can open calendar events in phone.

Dav5x ID and internal Android calendar ID is not matching for calendar events, so we need to provide uniqueness via depending on event name and event start date. 

Unified Search API doesn't provide solid date information for the calendar events, we can only get formatted date as String in subline section. We can add createdAt inside the attributes as UNIX Timestamp so that we can have exact date inside the response, and we can compare with internal Android calendar events without parsing the formatted date string.

**Current Implementation**

```
{
"thumbnailUrl": "",
"title": "Event 1",
"subline": "Jan 13, 2021",
"resourceUrl": "http://localhost/nc/index.php/apps/calendar/edit/L25jL3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy90b2JpL3BlcnNvbmFsL0ZEOTI5NTNDLTE2MTYtNDhDOC04NUQ3LUM1Q0Y1RkY3Nzg5Mi5pY3M=",
"icon": "icon-calendar-dark",
"rounded": false,
"attributes": []
}
```

**Suggestion**

```
{
"thumbnailUrl": "",
"title": "Event 1",
"subline": "Jan 13, 2021",
"resourceUrl": "http://localhost/nc/index.php/apps/calendar/edit/L25jL3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy90b2JpL3BlcnNvbmFsL0ZEOTI5NTNDLTE2MTYtNDhDOC04NUQ3LUM1Q0Y1RkY3Nzg5Mi5pY3M=",
"icon": "icon-calendar-dark",
"rounded": false,
"attributes": [
"createdAt": 1719225231
]
}
```

Is the subline variable formatted according to the user's locale settings, or can we get a consistent format regardless of locale settings?

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
fix https://github.com/nextcloud/contacts/issues/3997
fix https://github.com/nextcloud/calendar/issues/6079

<!--
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by alperozturk96** [comment]
@tobiasKaminsky Contact UUID is not matching with internal Android UUID synced by DAV5X therefore for contact I had to use displayName. Calendar events timestamps are correct and matching.

File: apps/dav/lib/Search/ContactsSearchProvider.php [hunk 1]
```diff
@@ -111,7 +111,13 @@ public function search(IUser $user, ISearchQuery $query): SearchResult {
 			$subline = $this->generateSubline($vCard);
 			$resourceUrl = $this->getDeepLinkToContactsApp($addressBook['uri'], (string) $vCard->UID);
 
-			return new SearchResultEntry($thumbnailUrl, $title, $subline, $resourceUrl, 'icon-contacts-dark', true);
+			$result = new SearchResultEntry($thumbnailUrl, $title, $subline, $resourceUrl, 'icon-contacts-dark', true);
+			$result->addAttribute("displayName", $title);
+			$result->addAttribute("email", $subline);
+			$result->addAttribute("phoneNumber", (string)$vCard->TEL);
+			$result->addAttribute("uid", (string) $vCard->UID);
```
Comment (tobiasKaminsky) [inline apps/dav/lib/Search/ContactsSearchProvider.php:118]: uid can be removed

**Review by alperozturk96 — APPROVED** [review]
Tested on client side working as expected

File: apps/dav/lib/Search/EventsSearchProvider.php [hunk 2]
```diff
@@ -157,8 +157,13 @@ public function search(
 				$calendar = $subscriptionsById[$eventRow['calendarid']];
 			}
 			$resourceUrl = $this->getDeepLinkToCalendarApp($calendar['principaluri'], $calendar['uri'], $eventRow['uri']);
+			$result = new SearchResultEntry('', $title, $subline, $resourceUrl, 'icon-calendar-dark', false);
 
-			return new SearchResultEntry('', $title, $subline, $resourceUrl, 'icon-calendar-dark', false);
+			$dtStart = $component->DTSTART;
+			$startDateTime = $dtStart->getDateTime()->format('U');
+			$result->addAttribute("createdAt", $startDateTime);
```
Comment (Altahrim) [inline apps/dav/lib/Search/EventsSearchProvider.php:164]: We can probably fix the notices like this:
```suggestion
			if ($dtStart instanceof Date) {
				$startDateTime = $dtStart->getDateTime()->format('U');
				$result->addAttribute("createdAt", $startDateTime);
			}
```

Comment (tobiasKaminsky) [reply apps/dav/lib/Search/EventsSearchProvider.php:164]: Which date is this?
> use Sabre\VObject\Property\ICalendar\Date;

did not work

**Review by Altahrim — APPROVED** [review]

**Pull request merged**